core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -309,36 +311,47 @@ private UpdateResultHolder executeUpdateRequest(UpdateRequest updateRequest, Ind
         for (int i = 0; i < request.items().length; i++) {
             BulkItemRequest item = request.items()[i];
             assert item.getPrimaryResponse() != null : "expected primary response to be set for item [" + i + "] request ["+ item.request() +"]";
-            if (item.getPrimaryResponse().isFailed() == false &&
-                    item.getPrimaryResponse().getResponse().getResult() != DocWriteResponse.Result.NOOP) {
+            if (item.getPrimaryResponse().getResponse().getResult() != DocWriteResponse.Result.NOOP) {
```

Comment:
I wonder (though not sure) if we can check for the existence of a seqNo and that's it.